server/datastore/mysql/migrations/tables/20211215110132_AddHostEmailsTable.go

```diff
@@ -0,0 +1,40 @@
+package tables
+
+import (
+	"database/sql"
+
+	"github.com/pkg/errors"
+)
+
+func init() {
+	MigrationClient.AddMigration(Up_20211215110132, Down_20211215110132)
+}
+
+func Up_20211215110132(tx *sql.Tx) error {
+	hostEmailsTable := `
+		CREATE TABLE IF NOT EXISTS host_emails (
+			id         int(10) unsigned NOT NULL AUTO_INCREMENT,
+			host_id    int(10) UNSIGNED NOT NULL,
+			email      varchar(255) NOT NULL,
+			source     varchar(255) NOT NULL,
```

Comment:
Do we need the autoincrement id? Could we instead have the primary key be `host_id, email, source`?

Reply:
That's a mighty long PK though, not recommended.

Reply:
Is it less recommended than the locking that happens with inserts and autoincrements?

Reply:
That sounds like a trick question :D I'm not a mysql expert, but at a previous place, we've had percona make an audit of our mysql schema, and we had some PKs that were a few dozen chars (don't remember exactly, maybe 100, 120 max) and they recommended making that a hash or using a different key as that was quite big for the index speed and space (basically what's mentioned here: https://stackoverflow.com/questions/13996565/is-there-a-recommended-size-for-a-mysql-primary-key). The locking-due-to-auto-increment issue we're seeing, I don't remember experiencing that before, I can't really tell what would be best here in our case but I know that would be a very big, inefficient PK space and speed-wise. That being said, if you think we'll end up debugging locks, I'm happy to switch it to the triplet!

Reply:
Well, it wasn't a question necessarily directed at you, more like an open question :)

We know how to measure this, so let's leave it as is and make sure we give it a good shake in the loadtest env.